octodns/provider/azuredns.py

```diff
@@ -538,8 +570,19 @@ def _check_zone(self, name, create=False):
         # If not, and its time to create, lets do it.
         if create:
             self.log.debug('_check_zone:no matching zone; creating %s', name)
-            create_zone = self._dns_client.zones.create_or_update
-            create_zone(self._resource_group, name, Zone(location='global'))
+
+            if (self._private):
+                zone_model = PrivateZone
+                create_method = 'begin_create_or_update'
+            else:
+                zone_model = Zone
+                create_method = 'create_or_update'
+
+            create_zone = attrgetter('_dns_client.{}.{}'.format(
+                                     self._zone_type, create_method))(self)
+            poller = create_zone(self._resource_group, name,
+                                 zone_model(location='global'))
+            self._private and poller.result()  # wait for zone creation
```

Comment:
Blocks on each zone creation. Less than ideal, obviously, but acceptable for now IMO.